vyper/functions/functions.py

```diff
@@ -350,6 +356,93 @@ def _sha3(expr, args, kwargs, context):
     )
 
 
+def _make_sha256_call(inp_start, inp_len, out_start, out_len):
+    return ['assert', [
+            'call',
+            ['gas'],  # gas
+            SHA256_ADDRESS,  # address
+            0,  # value
+            inp_start,
+            inp_len,
+            out_start,
+            out_len
+        ]
+    ]
+
+
+@signature(('str_literal', 'bytes', 'string', 'bytes32'))
+def sha256(expr, args, kwargs, context):
+    sub = args[0]
+    # Literal input
+    if isinstance(sub, bytes):
```

Comment:
I think this subroutine has different behavior for string and bytes literals.

Reply:
Minimal repro:
```
vyper -f ir /dev/stdin <<EOF
@public
def bar():
    x: bytes32 = sha256(b"test")
    y: bytes32 = sha256("test")
EOF
```

Reply:
Fixed, thanks ;)